src/main/java/edu/byu/cs/server/Server.java

```diff
@@ -23,99 +30,127 @@ public int start(int desiredPort) {
     }
 
     private int setupEndpoints(int port) {
-        port(port);
+        app = Javalin.create(config -> {
+                    config.staticFiles.add("/frontend/dist");
+
+                    config.jsonMapper(Serializer.jsonMapper);
+
+                    config.router.apiBuilder(() -> {
+                        before(provider.beforeAll());
 
-        webSocket("/ws", WebSocketController.class);
-        webSocketIdleTimeoutMillis(300000);
+                        path("/auth", () -> {
+                            get("/callback", provider.callbackGet());
+                            get("/login", provider.loginGet());
 
-        staticFiles.location("/frontend/dist");
+                            // TODO does Javalin guarantee this...?
+                            // all routes after this point require authentication
+                            post("/logout", provider.logoutPost());
+                        });
 
-        before(provider.beforeAll());
+                        path("/api", () -> {
+                            before("/*", ctx -> {
+                                if (ctx.method() != HandlerType.OPTIONS) provider.verifyAuthenticatedMiddleware().handle(ctx);
+                            });
 
```

Comment:
This verifyAuthenticatedMiddleware guarantees all the other endpoints are authenticated